internal/bgp/community/community.go

```diff
@@ -0,0 +1,158 @@
+// distinguish between extended and large communities.
+const largeBGPCommunityMarker = "large"
+
+// BGPCommunity represents a BGP community.
+type BGPCommunity interface {
+	LessThan(BGPCommunity) bool
+	String() string
+}
+
+// New parses the provided string and returns a newly created BGPCommunity object.
+// Strings are parsed according to Juniper style  syntax (https://www.juniper.net/documentation/us/en/software/\
+// junos/routing-policy/bgp/topics/concept/policy-bgp-communities-extended-communities-match-conditions-overview.html
+// Legacy communities are of format "<AS number>:<community value>".
+// Extended communities (support of which is yet to be implemented) are of format "<type>:<administrator>:<assigned-number>".
+// Large communities are of format large:<global administrator>:<localdata part 1>:<localdata part 2>.
+func New(c string) (BGPCommunity, error) {
+	var bgpCommunity BGPCommunity
+
+	fs := strings.Split(c, ":")
+	switch l := len(fs); l {
+	case 2:
+		a, err := strconv.ParseUint(fs[0], 10, 16)
+		if err != nil {
+			return bgpCommunity, fmt.Errorf("%w: invalid first section of community %q: %s",
+				ErrInvalidCommunityValue, fs[0], err)
+		}
+		b, err := strconv.ParseUint(fs[1], 10, 16)
+		if err != nil {
+			return bgpCommunity, fmt.Errorf("%w: invalid second section of community %q: %s",
+				ErrInvalidCommunityValue, fs[1], err)
+		}
+		return BGPCommunityLegacy((uint32(a) << 16) + uint32(b)), nil
+	case 4:
+		if fs[0] != largeBGPCommunityMarker {
+			return bgpCommunity, fmt.Errorf("%w: invalid marker for large community, expected community to be of "+
+				"format %s:<uint32>:<uint32>:<uint32> but got %s instead",
+				ErrInvalidCommunityValue, largeBGPCommunityMarker, fs)
+		}
+		var fields [3]uint32
+		for i := 1; i < 4; i++ {
+			b, err := strconv.ParseUint(fs[i], 10, 32)
+			if err != nil {
+				return bgpCommunity, fmt.Errorf("%w: invalid section %q of community, err: %q",
+					ErrInvalidCommunityValue, fs[i], err)
+			}
+			fields[i-1] = uint32(b)
+		}
+		return BGPCommunityLarge{
+			globalAdministrator: fields[0],
+			localDataPart1:      fields[1],
+			localDataPart2:      fields[2],
+		}, nil
+	}
+
+	return bgpCommunity, fmt.Errorf("%w: %s", ErrInvalidCommunityFormat, c)
+}
+
+// BGPCommunityLegacy holds the internal representation of a BGP legacy community.
+type BGPCommunityLegacy uint32
+
+// LessThan makes 2 different BGPCommunity objects comparable. For the sake of comparison, legacy communities are
+// considered to be large communities in format <legacy community>:0:0 and thus can be compared with large communities.
+func (b BGPCommunityLegacy) LessThan(c BGPCommunity) bool {
+	return lessThan(b, c)
+}
+
+// String returns the string representation of this community. Legacy communities will be printed in new-format,
+// "<AS number>:<community value>".
+func (b BGPCommunityLegacy) String() string {
+	return legacyCommunityToString(uint32(b))
```

Comment:
Would it make sense to move the function implementation here (without wrapping it)?

Reply:
done